zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -497,219 +528,201 @@ pub(super) const TABLE_SQLITE_SEQUENCE: &str = "CREATE TABLE sqlite_sequence(nam
+LEFT JOIN blocks ON blocks.height = notes.mined_height
 LEFT JOIN sent_note_counts
-        ON sent_note_counts.account_id = notes.account_id
-        AND sent_note_counts.txid = notes.txid
+     ON sent_note_counts.account_id = notes.account_id
+     AND sent_note_counts.txid = notes.txid
 GROUP BY notes.account_id, notes.txid";
 
 pub(super) const VIEW_TX_OUTPUTS: &str = "
 CREATE VIEW v_tx_outputs AS
-SELECT transactions.txid              AS txid,
-        v_received_notes.pool          AS output_pool,
-        v_received_notes.output_index  AS output_index,
-        sent_notes.from_account_id     AS from_account_id,
-        v_received_notes.account_id    AS to_account_id,
-        NULL                           AS to_address,
-        v_received_notes.value         AS value,
-        v_received_notes.is_change     AS is_change,
-        v_received_notes.memo          AS memo
-FROM v_received_notes
+SELECT transactions.txid            AS txid,
+       ro.pool                      AS output_pool,
+       ro.output_index              AS output_index,
+       sent_notes.from_account_id   AS from_account_id,
+       ro.account_id                AS to_account_id,
+       NULL                         AS to_address,
+       ro.value                     AS value,
+       ro.is_change                 AS is_change,
+       ro.memo                      AS memo
+FROM v_received_outputs ro
 JOIN transactions
-    ON transactions.id_tx = v_received_notes.tx
+    ON transactions.id_tx = ro.transaction_id
 LEFT JOIN sent_notes
-    ON sent_notes.id = v_received_notes.sent_note_id
-UNION
-SELECT utxos.prevout_txid           AS txid,
-        0                            AS output_pool,
-        utxos.prevout_idx            AS output_index,
-        NULL                         AS from_account_id,
-        utxos.received_by_account_id AS to_account_id,
-        utxos.address                AS to_address,
-        utxos.value_zat              AS value,
-        0                            AS is_change,
-        NULL                         AS memo
-FROM utxos
+    ON sent_notes.id = ro.sent_note_id
 UNION
 SELECT transactions.txid            AS txid,
-        sent_notes.output_pool       AS output_pool,
-        sent_notes.output_index      AS output_index,
-        sent_notes.from_account_id   AS from_account_id,
-        v_received_notes.account_id  AS to_account_id,
-        sent_notes.to_address        AS to_address,
-        sent_notes.value             AS value,
-        0                            AS is_change,
-        sent_notes.memo              AS memo
+       sent_notes.output_pool       AS output_pool,
+       sent_notes.output_index      AS output_index,
+       sent_notes.from_account_id   AS from_account_id,
+       ro.account_id                AS to_account_id,
+       sent_notes.to_address        AS to_address,
+       sent_notes.value             AS value,
+       0                            AS is_change,
```

Comment:
It just occurs to me, will this be correct behavior after ZIP 320 integration?

Reply:
I don't see why not, but we should test it. Note that `put_transparent_output` is called from `store_sent_tx` for ephemeral outputs.